### PR TITLE
feature/AKD-189: Added Azure 'Always On Enabled' Plugin and test cases

### PR DESCRIPTION
**Pull request by ali-imran7**

### CODE REVIEW

File: plugins/azure/appservice/alwaysOnEnabled.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
+var async = require('async');
+var helpers = require('../../../helpers/azure');
+
+module.exports = {
+    title: 'Web Apps Always On Enabled',
+    category: 'App Service',
+    description: 'Ensures that Azure Web Apps have Always On feature enabled.',
+    more_info: 'Keeps the app loaded even when there\'s no traffic. It\'s required for continuous WebJobs or for WebJobs that are triggered using a CRON expression.',
```
Comment (AkhtarAmir) [inline plugins/azure/appservice/alwaysOnEnabled.js:8]:
```suggestion
    more_info: 'Always On feature keeps the app loaded even when there\'s no traffic. It\'s required for continuous WebJobs or for WebJobs that are triggered using a CRON expression.',
```

File: plugins/azure/appservice/alwaysOnEnabled.js [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+    run: function(cache, settings, callback) {
+        var results = [];
+        var source = {};
+        var locations = helpers.locations(settings.govcloud);
+
+        async.each(locations.webApps, function(location, rcb) {
+            const webApps = helpers.addSource(
+                cache, source, ['webApps', 'list', location]
+            );
```
Comment (AkhtarAmir) [inline plugins/azure/appservice/alwaysOnEnabled.js:21]:
```suggestion
            const webApps = helpers.addSource(cache, source, ['webApps', 'list', location]);
```

File: plugins/azure/appservice/alwaysOnEnabled.js [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+                if (webApp && webApp.kind === 'functionapp') {
+                    helpers.addResult(results, 0, 'Always On feature can not be configured for the function App', location, webApp.id);
+                    return scb();
+                }
+
+                const configs = helpers.addSource(
+                    cache, source, ['webApps', 'listConfigurations', location, webApp.id]
+                );
+
+                if (!configs || configs.err || !configs.data) {
```
Comment (AkhtarAmir) [inline plugins/azure/appservice/alwaysOnEnabled.js:47]:
```suggestion
                if (!configs || configs.err || !configs.data || !configs.data.length) {
```

File: plugins/azure/appservice/alwaysOnEnabled.js [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+                    helpers.addResult(results, 3,
+                        'Unable to query for Web App Configs: ' + helpers.addError(configs), location);
+                    return scb();
+                }
+
+                if (!configs.data.length) {
+                    helpers.addResult(results, 0, 'No Web App configs found', location, webApp.id);
+                    return scb();
+                }
```
Comment (AkhtarAmir) [inline plugins/azure/appservice/alwaysOnEnabled.js:56]:
```suggestion
```

File: plugins/azure/appservice/alwaysOnEnabled.js [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+
+                configs.data.forEach(config => {
+                    if (config.alwaysOn) {
```
Comment (AkhtarAmir) [inline plugins/azure/appservice/alwaysOnEnabled.js:59]: Check this flag in find

**Review by AkhtarAmir — CHANGES_REQUESTED** [review]

File: plugins/azure/appservice/alwaysOnEnabled.js [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+                return rcb();
+            }
+
+            if (!webApps.data.length) {
+                helpers.addResult(results, 0, 'No existing Web Apps found', location);
+                return rcb();
+            }
+
+            async.each(webApps.data, function(webApp, scb) {
+                if (webApp && webApp.kind === 'functionapp') {
```
Comment (giorod3) [inline plugins/azure/appservice/alwaysOnEnabled.js:35]:
```suggestion
                if (webApp && webApp.kind && webApp.kind === 'functionapp') {
```

**Review by giorod3 — APPROVED** [review]

**Pull request merged**